src/integration/NotificationOnEditHandler.spec.ts

```diff
@@ -0,0 +1,241 @@
+import express from "express"
+import request from "supertest"
+
+import { NotificationOnEditHandler } from "@middleware/notificationOnEditHandler"
+
+import UserSessionData from "@classes/UserSessionData"
+
+import {
+  Notification,
+  Repo,
+  Reviewer,
+  ReviewMeta,
+  ReviewRequest,
+  ReviewRequestView,
+  Site,
+  SiteMember,
+  User,
+  Whitelist,
+} from "@database/models"
+import { generateRouterForUserWithSite } from "@fixtures/app"
+import {
+  mockEmail,
+  mockIsomerUserId,
+  mockSiteName,
+} from "@fixtures/sessionData"
+import { GitHubService } from "@services/db/GitHubService"
+import * as ReviewApi from "@services/db/review"
+import { ConfigYmlService } from "@services/fileServices/YmlFileServices/ConfigYmlService"
+import { getUsersService, notificationsService } from "@services/identity"
+import CollaboratorsService from "@services/identity/CollaboratorsService"
+import IsomerAdminsService from "@services/identity/IsomerAdminsService"
+import SitesService from "@services/identity/SitesService"
+import ReviewRequestService from "@services/review/ReviewRequestService"
+import { sequelize } from "@tests/database"
+
+const mockSiteId = "1"
+const mockSiteMemberId = "1"
+
+const mockGithubService = {
+  getPullRequest: jest.fn(),
+  getComments: jest.fn(),
+}
+const usersService = getUsersService(sequelize)
+const reviewRequestService = new ReviewRequestService(
+  (mockGithubService as unknown) as typeof ReviewApi,
+  User,
+  ReviewRequest,
+  Reviewer,
+  ReviewMeta,
+  ReviewRequestView
+)
+const sitesService = new SitesService({
+  siteRepository: Site,
+  gitHubService: (mockGithubService as unknown) as GitHubService,
+  configYmlService: (jest.fn() as unknown) as ConfigYmlService,
+  usersService,
+  isomerAdminsService: (jest.fn() as unknown) as IsomerAdminsService,
+  reviewRequestService,
+})
+const collaboratorsService = new CollaboratorsService({
+  siteRepository: Site,
+  siteMemberRepository: SiteMember,
+  sitesService,
+  usersService,
+  whitelist: Whitelist,
+})
+
+const notificationsHandler = new NotificationOnEditHandler({
+  reviewRequestService,
+  collaboratorsService,
+  sitesService,
+  notificationsService,
+})
+
+// Set up express with defaults and use the router under test
+const subrouter = express()
+const subSubrouter = express()
```

Comment:
Hey, is the creation of two routers intended?

Reply:
Yup - this is to simulate how this notification handler functions as closely as possible. The notification handler (represented by `subSubrouter`) is inserted into our `authenticatedSites` subrouter (represented by `subrouter` here), placed after any regular routes (represented by the `/:siteName/test` endpoint here) - this simulates our existing setup as closely as possible.

Reply:
haha could we rename this to `notificationHandlerRouter` and `authenticatedSitesRouter` instead?

Reply:
Oops correction on the above explanation - `subSubrouter` represents a router that the request is routed to and not the notification handler - the notification handler is just inserted in afterwards. As discussed offline - the names of the routers have been changed to make this clearer!